add a flag to determine if node has broadcasting

migeed-z opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61790**
* #61776

Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

migeed-z committed:
add a flag to determine if node has broadcasting

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"




[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"




[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"




[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29772587).

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

jamesr66a commented on torch/fx/experimental/graph_gradual_typechecker.py:
```diff
@@ -106,6 +106,13 @@ def add_inference_rule(n: Node):
     (new_t1, new_t2) = broadcast_types(t1, t2)
+
+    if new_t1 != t1:
+        n.args[0].broadcast = True  # type:ignore[attr-defined]
```
are `n.args[0]` and `args[1]` expected to always be `Node`s? Would probably be a good idea to add an assert to check that and then put the broacast flag in `node.meta` rather than monkey-patching a new attribute

migeed-z replied:
I fixed it and also the broadcasting part we talked about which was in the previous code. Let me know it looks reasonable!

jamesr66a reviewed (approved):
Looks good, just a small comment

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]

migeed-z committed:
Update on "add a flag to determine if node has broadcasting"


Differential Revision: [D29772587](https://our.internmc.facebook.com/intern/diff/D29772587)

[ghstack-poisoned]
